cpp/tests/interop/arrow_data_structures_test.cpp

```diff
@@ -0,0 +1,297 @@
+
+    auto [arrow_schema_from_nanoarrow_array, arrow_array_from_arrow_column] =
+      export_to_arrow(arrow_column_from_nanoarrow_array);
+    arrow_column_from_nanoarrow_array.to_arrow_schema(arrow_schema_from_nanoarrow_array.get());
+    arrow_column_from_nanoarrow_array.to_arrow(arrow_array_from_arrow_column.get(),
+                                               ARROW_DEVICE_CUDA);
+
+    auto arrow_column_from_arrow_array = cudf::interop::arrow_column(
+      std::move(*arrow_schema_from_nanoarrow_array), std::move(*arrow_array_from_arrow_column));
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(col.view(), *arrow_column_from_arrow_array.view());
+  }
+}
+
+TEST_F(ArrowColumnTest, ToFromHost)
+{
+  cudf::test::fixed_width_column_wrapper<int32_t> int_col{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}};
+  auto col = cudf::column(int_col);
+  auto arrow_column_from_cudf_column =
+    cudf::interop::arrow_column(std::move(col), cudf::interop::get_column_metadata(int_col));
+
+  auto [arrow_schema_from_arrow_column, arrow_array_from_arrow_column] =
+    export_to_arrow(arrow_column_from_cudf_column, ARROW_DEVICE_CPU);
+  arrow_column_from_cudf_column.to_arrow_schema(arrow_schema_from_arrow_column.get());
+  arrow_column_from_cudf_column.to_arrow(arrow_array_from_arrow_column.get(), ARROW_DEVICE_CPU);
+
+  auto arrow_column_from_arrow_array = cudf::interop::arrow_column(
+    std::move(*arrow_schema_from_arrow_column), std::move(*arrow_array_from_arrow_column));
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(int_col, *arrow_column_from_arrow_array.view());
+}
+
+struct ArrowTableTest : public cudf::test::BaseFixture {};
+
+TEST_F(ArrowTableTest, TwoWayConversion)
+{
+  cudf::test::fixed_width_column_wrapper<int32_t> int_col{{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}};
+  cudf::test::fixed_width_column_wrapper<float> float_col{
+    {1., 2., 3., 4., 5., 6., 7., 8., 9., 10.}};
+  auto original_view = cudf::table_view{{int_col, float_col}};
+  cudf::table table{cudf::table_view{{int_col, float_col}}};
+  auto arrow_table_from_cudf_table =
+    cudf::interop::arrow_table(std::move(table), cudf::interop::get_table_metadata(original_view));
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(original_view, *arrow_table_from_cudf_table.view());
+
+  auto [arrow_schema_from_arrow_table, arrow_array_from_arrow_table] =
+    export_to_arrow(arrow_table_from_cudf_table);
+  arrow_table_from_cudf_table.to_arrow_schema(arrow_schema_from_arrow_table.get());
+  arrow_table_from_cudf_table.to_arrow(arrow_array_from_arrow_table.get(), ARROW_DEVICE_CUDA);
+
+  auto arrow_table_from_arrow_array = cudf::interop::arrow_table(
+    std::move(*arrow_schema_from_arrow_table), std::move(*arrow_array_from_arrow_table));
+  CUDF_TEST_EXPECT_TABLES_EQUAL(original_view, *arrow_table_from_arrow_array.view());
+}
+
+TEST_F(ArrowTableTest, ComplexNanoarrowDeviceTables)
+{
+  auto [tbl, schema, arr] = get_nanoarrow_tables(100);
+  ArrowDeviceArray device_arr{
+    .array       = {},
+    .device_id   = 0,
+    .device_type = ARROW_DEVICE_CUDA,
```

Comment:
is it okay to (already) use a C++20 language feature?

Reply:
If the compiler does not complain, neither will I :) I'm mostly surprised this compiles with the current version(s).